### PR TITLE
Fix typos, grammar in docs

### PR DESCRIPTION
**Pull request by kiike**
This work-in-progress PR fixes the typos and grammar in the docs and make the docs somewhat easier to read.

### CODE REVIEW

**Comment by psychon** [comment]
Thanks! 👍

**Comment by kiike** [comment]
Hi @psychon! Thanks for the feeback! I'm having a look at the other ones as we speak.

File: docs/05-awesomerc.md.lua [hunk 1]
```diff
@@ -30,27 +30,28 @@ Here is the modules being imported:
 
 sections.DOC_ERROR_HANDLING = [[
 Awesome is a window managing framework. It allows its users great (ultimate?)
-flexibility. However, it also allows the user to write invalid code. There is
-multiple "levels" of problems:
+flexibility. However, it also allows the user to write invalid code. Here's a
+non-extensive list of possible errors:
```
Comment (kiike) [inline docs/05-awesomerc.md.lua]: Yep!

**Comment by kiike** [comment]
It looks good to go :)

File: docs/03-declarative-layout.md [hunk 2]
```diff
@@ -17,15 +17,15 @@ Awesome provide 2 collections of widgets:
 
 ### Containers
 
-Containers are widget wrapping another widget. It can be used to add decorations
-or to modify the content of the child widget.
+A container is a widgets that wraps another widget. It can be used to add
```
Comment (blueyed) [inline docs/03-declarative-layout.md]: s/widgets/widget/

File: docs/03-declarative-layout.md [hunk 3]
```diff
@@ -106,9 +106,9 @@ Result:
 ![Example2 screenshot](../images/widgetlayout1.png)
 
 
-### Use an `wibox.layout.align` layout
+### Use a `wibox.layout.align` layout
 The `wibox.layout.align` is a little different. While most layouts will
-ignore any `nil` lines, the `align` layout rely on them so `left`, `middle`
+ignore any `nil` lines, the `align` layout relies on them so `left`, `middle`
 and `right` can be defined
```
Comment (blueyed) [inline docs/03-declarative-layout.md]: Missing dot at the end.

File: docs/03-declarative-layout.md [hunk 4]
```diff
@@ -177,23 +177,23 @@ Code:
 ### Accessing widgets
 
 For each widget or container, it is possible to add an `identifier` attribute
-so the widget can be accessed later.
+so that it accessed later.
```
Comment (blueyed) [inline docs/03-declarative-layout.md]: s/it/it can be/ (or similar)

File: docs/03-declarative-layout.md [hunk 5]
```diff
@@ -177,23 +177,23 @@ Code:
...
-The first method mixes the imperative and declarative syntax, but makes the code
+The first method mixes the imperative and declarative syntax, and makes the code
 less readable. The second is a little verbose and only works if every node in
-the chain have a valid identifier. The last one doesn't require long paths,
+the chain has a valid identifier. The last one doesn't require long paths,
 but it is not easy to get a specific instance if multiple widgets have the
 same identifier.
 
-WARNING: The widget identifier must not use reseved name. This include all
+WARNING: The widget identifier must not use reserved name. This includes all
```
Comment (blueyed) [inline docs/03-declarative-layout.md]: s/name/names/

File: docs/03-declarative-layout.md [hunk 6]
```diff
@@ -177,23 +177,23 @@ Code:
...
 method names, existing widget attributes, `layout` and `widget`. Names should
-also respect the lua variable name policies (case sensitive, alphanumeric and
+also respect the Lua variable conventions (case-sensitive, alphanumeric,
 underscore characters and non-numeric first character)
```
Comment (blueyed) [inline docs/03-declarative-layout.md]: Dot at the end.

File: docs/05-awesomerc.md.lua [hunk 7]
```diff
@@ -30,27 +30,28 @@ Here is the modules being imported:
...
    the best way to debug this is either using `print()` or using `gdb`. For
    this, see the [Debugging tips Readme section](../documentation/01-readme.md.html)
- * Deprecated APIs: Awesome API is not frozen for eternity. While after a
-   decade and recent changes to enforce consistency, it doesn't change as much,
-   it will likely be changed in the future. When possible, changes wont cause
-   errors but will instead print a deprecation message in Awesome logs. Those
-   logs are placed in various places depending on the distribution. By default,
-   Awesome will print this on stderr and stdout.
+ * Deprecated APIs: The Awesome API is not frozen for eternity. After a decade
+   of development and recent changes to enforce consistency, it doesn't hasn't
```
Comment (blueyed) [inline docs/05-awesomerc.md.lua]: "doesn't hasn't" is wrong.

File: docs/16-using-cairo.md [hunk 8]
```diff
@@ -1,7 +1,7 @@
 # Using Cairo and LGI
 
-These days, Awesome interface is mostly based on a library called LGI. It allows
-to access C libraries such as GTK, GLib, Cairo, Pango, PangoCairo and RSVG using
+These days, Awesome interface is mostly based on a library called LGI. It provides
```
Comment (blueyed) [inline docs/16-using-cairo.md]: s/Awesome/Awesome's/

File: docs/16-using-cairo.md [hunk 9]
```diff
@@ -1,7 +1,7 @@
...
+access C libraries such as GTK, GLib, Cairo, Pango, PangoCairo and RSVG using
```
Comment (blueyed) [inline docs/16-using-cairo.md]: s/access/access to/

**Comment by kiike** [comment]
Thanks for having a look at it, @psychon, @Elv13 and @blueyed! I can't believe how those errors could get throuh. They are fixed now as per your review.

**Comment by Elv13** [comment]
Thanks

**Comment by blueyed** [comment]
> throuh

No spell checking enabled maybe?!

Thanks a lot indeed!

**Comment by blueyed** [comment]
@Elv13 
It should have been squashed into a single commit!

**Pull request merged**